e2e/testinfra/terraform/README.md

```diff
@@ -11,7 +11,6 @@ Follow the instructions to [install the Terraform CLI].
 
 # Dev Environments
 
-The Terraform config for dev environments is defined in the [dev directory].
```

Comment:
there's still tf config to set up the project and network, right?

Reply:
maybe add a section about setting up the env with tf?

Reply:
Yes, there's just nothing specific in `dev/` anymore - as that was only used for setting up clusters before. Everything that is shared between dev and prow environment is in `common/` now, with some extra stuff for prow under `prow/`

Reply:
This section should still contain accurate instructions for setting up a new dev and/or prow environment